lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -564,13 +611,48 @@ def _update_relation_data(self, relation_id: int, data: dict) -> None:
             if relation:
                 relation.data[self.charm.model.app].update(data)
 
+    def _is_mongos_reachable(self) -> bool:
+        """Returns True if mongos is reachable."""
+        if not self.model.get_relation(self.relation_name):
+            logger.info("Mongos is not reachable, no relation to config-sever")
+            return False
+
+        mongos_hosts = self.get_mongos_hosts()
+        self.charm.remote_mongos_config(set(mongos_hosts))
+        config = self.charm.remote_mongos_config(set(mongos_hosts))
+
+        # use a URI that is not dependent on the operator password, as we are not guaranteed that
+        # the shard has received the password yet.
+        uri = f"mongodb://{','.join(mongos_hosts)}"
+        with MongosConnection(config, uri) as mongo:
+            return mongo.is_ready
+
+    def _is_added_to_cluster(self) -> bool:
+        """Returns True if the shard has been added to the cluster."""
+        return json.loads(self.charm.app_peer_data.get("added_to_cluster", "False"))
+
+    def _is_shard_aware(self) -> bool:
+        """Returns True if shard is in cluster and shard aware."""
+        if not self.model.get_relation(self.relation_name):
+            logger.info(
+                "Mongos is not reachable, no relation to config-sever, cannot check shard status."
+            )
+            return False
+
+        mongos_hosts = self.get_mongos_hosts()
+        with MongosConnection(self.charm.remote_mongos_config(set(mongos_hosts))) as mongo:
+            return mongo.is_shard_aware(shard_name=self.charm.app.name)
+
     def has_config_server(self) -> bool:
         """Returns True if currently related to config server."""
         return len(self.charm.model.relations[self.relation_name]) > 0
 
-    def get_related_config_server(self) -> List[str]:
+    def get_related_config_server(self) -> str:
         """Returns the related config server."""
-        return [rel.app.name for rel in self.charm.model.relations[self.relation_name]]
+        config_server = [rel.app.name for rel in self.charm.model.relations[self.relation_name]]
```

Comment:
because we always return first element of the list, why do we need to build the list?
why not something like `self.charm.model.relations[self.relation_name][0].app.name`? 😄

Reply:
ah, we should do some error checking, so let me add that, then add your suggestion